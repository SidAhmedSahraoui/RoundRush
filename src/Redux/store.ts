import { Action, ThunkAction, configureStore } from "@reduxjs/toolkit";

import auth from "./auth/auth-slice";


export const store = configureStore({
  reducer: {
    auth
  },
  devTools: true,
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
