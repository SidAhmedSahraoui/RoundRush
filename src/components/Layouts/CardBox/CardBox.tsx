import React, { ReactNode } from "react";
import { createStyles, makeStyles } from "@mui/styles";
import Card from "@mui/material/Card";
import CardContent from "@mui/material/CardContent";


const useStyles = makeStyles(() =>
  createStyles({
    root: {
      textAlign: "center",
      width: "539px",
      left: "0px",
      top: "0px",
      borderRadius: "6px !important",
      margin: "auto",
      background: "rgba(255, 255, 255, 1)",
      padding: "8px",
      boxShadow: "0px 2px 10px 0px rgba(0, 0, 0, 0.1) !important",
      marginTop: "48px",
    }
  }),
);

interface CardBoxProps {
  children: ReactNode;
}

export const CardBox: React.FC<CardBoxProps> = ({ children }) => {
  const classes = useStyles();

  return (
    <Card className={classes.root}>
      <CardContent sx={{padding: "0px"}}>{children}</CardContent>
    </Card>
  );
};
