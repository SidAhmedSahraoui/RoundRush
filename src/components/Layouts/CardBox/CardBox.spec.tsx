/**
 * @jest-environment jsdom
 */

 import React from "react";
 import { cleanup, render } from "@testing-library/react";
 import { Provider } from "react-redux";
 import { store } from "../../../redux/store";
 import { CardBox } from "./CardBox";
 
 describe("CardBox", () => {
   afterEach(() => cleanup());
 
   test("Render CardBox component", () => {
     const view = render(
       <Provider store={store}>
           <CardBox>Test</CardBox>
       </Provider>
     );
 
     expect(view.container).toMatchSnapshot();
   });
 });
 