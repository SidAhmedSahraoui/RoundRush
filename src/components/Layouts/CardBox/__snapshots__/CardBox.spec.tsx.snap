// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardBox Render CardBox component 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root makeStyles-root-1 css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardContent-root css-1i9wt8r-MuiCardContent-root"
    >
      Test
    </div>
  </div>
</div>
`;

exports[`src/components/cards/card-box/index.tsx Render CardBox component 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root makeStyles-root-1 css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardContent-root css-1i9wt8r-MuiCardContent-root"
    >
      Test
    </div>
  </div>
</div>
`;
